lib/pyspark_matrix_similarity.py

```diff
@@ -35,3 +37,56 @@ def cos(x):
     df = df.withColumn("cos_similarity", my_udf(f.col("features")))
 
     return df.select(["id", "cos_similarity"])
+
+
+def intersection_over_union_for_row(
+    df: DataFrame,
+    movie_id: str,
+    column_name: str = "ludzie_filmu",
+) -> DataFrame:
+    """This function returns a DataFrame that contains intersection_over_union calculations for the given movie_id.
+
+    Example:
+            input Dataframe
+        +---------+-----------------------------+
+        |       id|          ludzie_filmu       |
+        +---------+-----------------------------+
+        |tt0000001|       ["a", "b", "c", "d"]  |
+        |tt0000003|       ["d", "e", "f", "g"]  |
+        +---------+-----------------------------+
+
+        intersection_over_union_for_row(df, "tt0000001", column_name = "ludzie_filmu" )
+
+            output
+
+        +---------+---------------+
+        |       id|     IoU       |
+        +---------+---------------+
+        |tt0000001|       1       |
+        |tt0000003|     0.142     |
+        +---------+---------------+
+
+
+
+    :param column_name:     String
+    :param df:              pyspark.sql.DataFrame
+    :param movie_id:        String
+    :return:                pyspark.sql.DataFrame"""
+
+    if not ("id" in df.columns and column_name in df.columns):
```

Comment:
To chyba trzeba odświeżyć względem nowych parametrów.

Reply:
Column name odświeżyłem ( wcześniej- ludzie filmu), mogę dodać drugi parametr do id, ale wszędzie tak nazywamy tę kolumnę, więc chyba nie ma potrzeby.

Reply:
Proszę dodaj parametr i asserty. Jest to analogiczna problematyka jak z poprzednim parametrem.